packages/opentelemetry-api/src/trace/span.ts

```diff
@@ -114,4 +115,12 @@ export interface Span {
    *     with the `AddEvent` operation and attributes using `setAttributes`.
    */
   isRecording(): boolean;
+
+  /**
+   * Sets exception as a span event
+   * @param exception the exception the only accepted values are string or Error
+   * @param [time] the time to set as Span's event time. If not provided,
+   *     use the current time.
+   */
+  recordException(exception: Exception, time?: TimeInput): void;
```

Comment:
We should also allow `string` since it is very common for error callbacks to accept strings.

Reply:
`Exception` is already a type of string too